awx/main/notifications/webhook_backend.py

```diff
@@ -32,8 +38,15 @@ def send_messages(self, messages):
         sent_messages = 0
         if 'User-Agent' not in self.headers:
             self.headers['User-Agent'] = "Tower {}".format(get_awx_version())
+        if self.http_method.lower() not in ['put','post']:
+            raise ValueError("HTTP method must be either 'POST' or 'PUT'.")
+        chosen_method = getattr(requests, self.http_method.lower(), None)
         for m in messages:
-            r = requests.post("{}".format(m.recipients()[0]),
+            auth = None
+            if self.username:
```

Comment:
Here it should be `if self.username or self.password`. After verification, all following four scenarios are valid

* `http://foo:password@url/`
* `http://foo@url/`
* `http://:password@url/`
* `http://url/`

Reply:
Is "password only" a thing?   @jakemcdermott's UI validation here doesn't follow the same pattern:  https://github.com/ansible/awx/pull/4124/commits/64a666bfdd53a2291f22bf4ad0abdde9c6e936bf

While basic auth with a username and no password is definitely a thing, I'm somewhat inclined to suggest that basic auth _without_ a username is pretty weird, and I dunno if it's a thing people actually want to do in practice.

Reply:
Problem here is that protocol allows it. And if protocols allows it, you never know how any service might be using it. If its possible (by protocol) let's make sure we support it, and this way we don't really care about how people deal with it.